src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -95,6 +95,11 @@ public struct ComActivationContext
         public static unsafe ComActivationContext Create(ref ComActivationContextInternal cxtInt)
         {
 #if FEATURE_COMINTEROP_UNMANAGED_ACTIVATION
+            if (!Marshal.IsComSupported)
```

Comment:
All of the managed APIs which will throw if COM is not supported should have `RequiresUnreferencedCodeAttribute` on them. We can do that in a separate change if you want, but without it, the experience is not right (if I enable COM and trimming I SHOULD get warnings - that's how we do it everywhere else).